sklearn/datasets/openml.py

```diff
@@ -61,8 +62,9 @@ def _open_openml_url(openml_path, data_home):
             pass
 
         try:
-            with gzip.GzipFile(local_path, 'wb') as fdst:
-                fsrc = urlopen(_OPENML_PREFIX + openml_path)
+            with open(local_path, 'wb') as fdst:
+                req.add_header('Accept-encoding', 'gzip')
+                fsrc = urlopen(req)
```

Comment:
This should probably double check that the http response says it is actually gzipped.

Reply:
When the responses' s `Content-Encoding` is not `gzip`, I see two options to handle this:

1. Do another request for without `gzip`.
2. Raise an exception.

Which do you prefer?